### PR TITLE
PYTHON-2057 Make 'name' a required argument for DriverInfo class

### PR DESCRIPTION
**Pull request by juliusgeo**
https://jira.mongodb.org/browse/PYTHON-2057

### CODE REVIEW

**Review by prashantmital — APPROVED** [review]
LGTM but please wait for Shane's approval before merging :).

**Review by ShaneHarvey — APPROVED** [review]

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Can you update the changelog and the migrate-to-pymongo4 document to mention this breaking change?

File: doc/migrate-to-pymongo4.rst
Comment (ShaneHarvey) [inline doc/migrate-to-pymongo4.rst:732]: I think there needs to be a blank line after the `.......` header.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/migrate-to-pymongo4.rst
Comment (juliusgeo) [reply doc/migrate-to-pymongo4.rst:733]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**
